### PR TITLE
[iov-bns] Some cleanups

### PR DESCRIPTION
**Pull request by webmaster128**
Some cleanups done before #493. All in here does is not a breaking change and can safely be merged into 0.8.x

### CODE REVIEW

File: packages/iov-bns/scripts/weave.sh [hunk 1]
```diff
@@ -1,21 +1,25 @@
 #!/bin/bash
+set -o errexit -o nounset -o pipefail
```
Comment (ethanfrey) [inline packages/iov-bns/scripts/weave.sh]: Your bash is cleaner than mine :+1:

File: packages/iov-bns/src/encode.spec.ts [hunk 2]
```diff
@@ -1,12 +1,10 @@
 import { Ed25519, Ed25519Keypair, Sha512 } from "@iov/crypto";
 
 import * as codecImpl from "./codecimpl";
-import { buildMsg, buildSignedTx, buildUnsignedTx } from "./encode";
-import { encodePrivKey, encodePubKey, encodeToken } from "./types";
-import { appendSignBytes, keyToAddress } from "./util";
+import { buildMsg, buildSignedTx, buildUnsignedTx, encodePrivkey, encodePubkey, encodeToken } from "./encode";
```
Comment (ethanfrey) [inline packages/iov-bns/src/encode.spec.ts]: Proper capitalization, right?

File: packages/iov-bns/src/encode.ts [hunk 3]
```diff
@@ -10,26 +11,55 @@ import {
...
 } from "@iov/bcp-types";
+import { Algorithm, PrivateKeyBundle, PublicKeyBundle } from "@iov/tendermint-types";
 
 import * as codecImpl from "./codecimpl";
-import { encodeFullSig, encodeToken } from "./types";
+import { encodeFullSig } from "./types";
 import { decodeBnsAddress, keyToAddress, preimageIdentifier } from "./util";
 
-export const buildSignedTx = (tx: SignedTransaction): codecImpl.app.ITx => {
+export function encodePubkey(publicKey: PublicKeyBundle): codecImpl.crypto.IPublicKey {
```
Comment (ethanfrey) [inline packages/iov-bns/src/encode.ts]: Bye, bye top-level lambdas.

I know, we agreed this needed to go

Comment (webmaster128) [reply packages/iov-bns/src/encode.ts]: There are some places in this PR where being forced to set return type adds real value to the code. This changes from _some interface_ to _the type we want to work with_

* https://github.com/iov-one/iov-core/pull/509/commits/6cdeee9f2fcadac7813602935e15fac9cac12ba4#diff-3a78df7aabe6b2e904eee5e9b98e2d19R46
* https://github.com/iov-one/iov-core/pull/509/commits/6cdeee9f2fcadac7813602935e15fac9cac12ba4#diff-3a78df7aabe6b2e904eee5e9b98e2d19R55

**Pull request merged**